### PR TITLE
docs: adds a description of cc modes to the overview

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation change**
Updates the Cruise Control summary in the [Strimzi Overview guide](https://strimzi.io/docs/operators/in-development/overview.html) to describe the new modes that can be used when scaling up/down. 

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

File: documentation/modules/overview/con-metrics-overview-tools.adoc [hunk 1]
```diff
@@ -59,5 +59,11 @@ Cruise Control collects resource utilization information to model and analyze th
 Based on _optimization goals_ that have been defined, Cruise Control generates _optimization proposals_ outlining how the cluster can be effectively rebalanced.
 When an _optimization proposal_ is approved, Cruise Control applies the rebalancing outlined in the proposal.
 
+You can generate optimization proposals in specific modes. 
+A `full` mode rebalance across all brokers is used by default.
+Other modes accommodate changes when scaling a cluster up or down.
+Generate an optimization proposal in `add-brokers` mode to include any newly added brokers in the rebalance.
```
Comment (tomncooper) [inline documentation/modules/overview/con-metrics-overview-tools.adoc:65]:
```suggestion
Generate an optimization proposal in `add-brokers` mode to move data from existing brokers to newly added ones.
```

File: documentation/modules/overview/con-metrics-overview-tools.adoc [hunk 2]
```diff
@@ -59,5 +59,11 @@ Cruise Control collects resource utilization information to model and analyze th
...
+Generate an optimization proposal in `remove-brokers` mode to exclude any brokers from the rebalance that are set to be removed.
```
Comment (tomncooper) [inline documentation/modules/overview/con-metrics-overview-tools.adoc:66]:
```suggestion
Generate an optimization proposal in `remove-brokers` mode to move all data off brokers to be removed, before scaling down.
```

**Review by tomncooper — CHANGES_REQUESTED** [review]
A few tweaks.

**Comment by scholzj** [comment]
@tomncooper Does this look good now?

**Review by tomncooper — APPROVED** [review]
LGTM

**Pull request merged**